File: apk-info.js
#!/usr/bin/env node

const fs = require('fs').promises;
const ApkReader = require('adbkit-apkreader');

/**
 * Extract information from an APK file
 * 
 * @param {string} apkPath - Path to the APK file
 * @returns {Promise<object|null>} Dictionary containing APK information
 */
async function extractApkInfo(apkPath) {
    try {
        console.log(`Analyzing APK: ${apkPath}`);
        
        // Parse APK using adbkit-apkreader
        const reader = await ApkReader.open(apkPath);
        const manifest = await reader.readManifest();
        console.log("Successfully parsed APK");
        
        // Extract basic information
        const info = {};
        
        console.log("Extracting package name...");
        info.package_name = manifest.package;
        
        console.log("Extracting version info...");
        info.version_name = manifest.versionName;
        info.version_code = manifest.versionCode?.toString();
        
        console.log("Extracting SDK versions...");
        info.min_sdk_version = manifest.usesSdk?.minSdkVersion?.toString();
        info.target_sdk_version = manifest.usesSdk?.targetSdkVersion?.toString();
        
        console.log("Extracting app name...");
        // Get the first application label as app name
        info.app_name = manifest.application.label;
        
        console.log("Extracting permissions...");
        info.permissions = manifest.usesPermissions?.map(p => p.name) || [];
        
        // Print information
        console.log("\nAPK Information:");
        console.log("-".repeat(50));
        console.log(`Package Name: ${info.package_name}`);
        console.log(`Version Name: ${info.version_name}`);
        console.log(`Version Code: ${info.version_code}`);
        console.log(`Min SDK Version: ${info.min_sdk_version}`);
        console.log(`Target SDK Version: ${info.target_sdk_version}`);
        console.log(`App Name: ${info.app_name}`);
        console.log("\nPermissions:");
        info.permissions.sort().forEach(permission => {
            console.log(`- ${permission}`);
        });
        
        return info;
        
    } catch (error) {
        console.error(`Error analyzing APK: ${error.message}`);
        console.error(error.stack);
        return null;
    }
}

// Main execution
if (require.main === module) {
    const args = process.argv.slice(2);
    if (args.length > 0) {
        const apkPath = args[0];
        extractApkInfo(apkPath).catch(error => {
            console.error("Failed to extract APK info:", error);
            process.exit(1);
        });
    } else {
        console.log("Please provide path to APK file");
        process.exit(1);
    }
}

module.exports = extractApkInfo;

File: apk-info.py
import sys
from androguard.misc import AnalyzeAPK

def extract_apk_info(apk_path):
    """
    Extract information from an APK file
    
    Args:
        apk_path (str): Path to the APK file
        
    Returns:
        dict: Dictionary containing APK information
    """
    try:
        print(f"Analyzing APK: {apk_path}")
        # Parse APK using androguard
        a, d, dx = AnalyzeAPK(apk_path)
        print("Successfully parsed APK")
        
        # Extract basic information
        info = {}
        
        print("Extracting package name...")
        info['package_name'] = a.get_package()
        
        print("Extracting version info...")
        info['version_name'] = a.get_androidversion_name()
        info['version_code'] = a.get_androidversion_code()
        
        print("Extracting SDK versions...")
        info['min_sdk_version'] = a.get_min_sdk_version()
        info['target_sdk_version'] = a.get_target_sdk_version()
        
        print("Extracting app name...")
        info['app_name'] = a.get_app_name()
        
        print("Extracting permissions...")
        info['permissions'] = a.get_permissions()
        
        # Print information
        print("\nAPK Information:")
        print("-" * 50)
        print(f"Package Name: {info['package_name']}")
        print(f"Version Name: {info['version_name']}")
        print(f"Version Code: {info['version_code']}")
        print(f"Min SDK Version: {info['min_sdk_version']}")
        print(f"Target SDK Version: {info['target_sdk_version']}")
        print(f"App Name: {info['app_name']}")
        print("\nPermissions:")
        for permission in sorted(info['permissions']):
            print(f"- {permission}")
            
        return info
        
    except Exception as e:
        print(f"Error analyzing APK: {e}")
        import traceback
        traceback.print_exc()
        return None

if __name__ == '__main__':
    if len(sys.argv) > 1:
        apk_path = sys.argv[1]
        extract_apk_info(apk_path)
    else:
        print("Please provide path to APK file")